signal.c

```diff
@@ -520,6 +520,38 @@ rb_f_kill(int argc, const VALUE *argv)
     return INT2FIX(i-1);
 }
 
+/*
+ *  call-seq:
+ *     Process.kill(pid)    -> boolean
+ *
+ * Checks the given process to see whether it is alive.
+ */
+VALUE
+rb_f_exists(VALUE self, VALUE val_pid)
+{
+    int pid;
+    switch (TYPE(val_pid)) {
```

Comment:
It is strongly advised that you duck type to avoid `TYPE()` macro.  To do so there is `rb_to_int()` function.  Use it instead.